src/Interpreters/Cache/FileCacheSettings.cpp

```diff
@@ -34,6 +34,8 @@ void FileCacheSettings::loadFromConfig(const Poco::Util::AbstractConfiguration &
     cache_on_write_operations = config.getUInt64(config_prefix + ".cache_on_write_operations", false);
     enable_filesystem_query_cache_limit = config.getUInt64(config_prefix + ".enable_filesystem_query_cache_limit", false);
     enable_cache_hits_threshold = config.getUInt64(config_prefix + ".enable_cache_hits_threshold", REMOTE_FS_OBJECTS_CACHE_ENABLE_HITS_THRESHOLD);
+    enable_limit_download_cache_size = config.getUInt64(config_prefix + ".enable_limit_download_cache_size", false);
```

Comment:
may be a better name for these settings could be `enable_bypass_cache_threashold` and `bypass_cache_threshold`? or something similar.. Because `limit_download_cache_size` can be confused with similar settings.

Reply:
OK, now using **enable_bypass_cache_with_threashold** and **bypass_cache_threashold**

Reply:
Just my two cents on settings names, maybe even better will be to use `max_object_to_cache` or similar?
Plus it seems that one setting is enough, if non zero, then enabled.